examples/compiler/01_iet.ipynb

```diff
@@ -0,0 +1,664 @@
+    "\n",
+    "At this point, the same `Operator` can be used for a completely different run. Let's create another `TimeFunction` for governing a similar domain. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Data([[[0., 0., 0.],\n",
+       "       [0., 0., 0.],\n",
+       "       [0., 0., 0.]],\n",
+       "\n",
+       "      [[0., 0., 0.],\n",
+       "       [0., 0., 0.],\n",
+       "       [0., 0., 0.]]], dtype=float32)"
+      ]
+     },
+     "execution_count": 29,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "u2 = TimeFunction(name='u', grid=grid)\n",
+    "u2.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Any of the operator default arguments may be replaced just by passing suitable key-value parameters.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "=========================================================================================\n",
+      "=========================================================================================\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Data([[[0., 0., 0.],\n",
+       "       [4., 4., 0.],\n",
+       "       [4., 4., 0.]],\n",
+       "\n",
+       "      [[0., 0., 0.],\n",
+       "       [3., 3., 0.],\n",
+       "       [3., 3., 0.]]], dtype=float32)"
+      ]
+     },
+     "execution_count": 30,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "op.apply(u=u2, x_m=1, x_M=2, y_m=0, y_M=1, time_M=3)\n",
+    "u2.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note, however, that there is no need for recompilation. Just-in-time (JIT) compilation occurs only once, triggered by the first execution.\n",
+    "\n",
+    "In summary, an `Operator` takes as input an ordered sequence of `SymPy` equations and performs _indexification_, _substitution_ and _domain-alignment_ for building _lowered equations_. The lowered equations are analyzed to collect information relevant for the `Operator` construction and execution, resulting in `Clusters` of ordered sequences of equations having the same _iteration space_ (`ISpace`) and no dimension-carried anti-dependences among them. Those `Clusters` go through _symbolic-optimization_ by the Devito Symbolic Engine (DSE), which consists of a series of passes, ranging from standard commmon sub-expression elimination (CSE) to more advanced rewrite procedures applied to each `Cluster`. The result is a new ordered sequence of `Clusters` that are lowered to an Iteration/Expression Tree (IET).\n",
```

Comment:
- `Devito` equation

>_indexification_, _substitution_ and _domain-alignment_ for building _lowered equations_.

details

Very dense. WOuld look at @fablup paper or at list put ref. This is hard to follow by itself.